apps/entity-service/src/bounding-boxes/bounding-box.service.ts

```diff
@@ -0,0 +1,271 @@
+import { Model, ModelStatic, Op, Sequelize, WhereOptions } from "sequelize";
+import { DataApiService } from "@terramatch-microservices/data-api";
+import { ConfigService } from "@nestjs/config";
+
+enum EntityType {
+  POLYGON = "Polygon",
+  SITE = "Site",
+  PROJECT = "Project",
+  LANDSCAPE = "Landscape",
+  COUNTRY = "Country"
+}
+
+interface BoundingBoxCoordinates {
+  minLng: number;
+  minLat: number;
+  maxLng: number;
+  maxLat: number;
+}
+
+@Injectable()
+export class BoundingBoxService {
+  constructor(private readonly dataApiService: DataApiService, private readonly configService: ConfigService) {}
+
+  private createBoundingBoxDto(minLng: number, minLat: number, maxLng: number, maxLat: number): BoundingBoxDto {
+    return new BoundingBoxDto({
+      bbox: [minLng, minLat, maxLng, maxLat]
+    });
+  }
+
+  private extractBoundingBoxFromEnvelopes(envelopes: Model[]): BoundingBoxCoordinates {
+    let maxLng = -Infinity;
+    let minLng = Infinity;
+    let maxLat = -Infinity;
+    let minLat = Infinity;
+
+    for (const envelope of envelopes) {
+      const geojson = JSON.parse(envelope.get("envelope") as string);
+      const coordinates = geojson.coordinates[0];
+
+      for (const point of coordinates) {
+        const [lng, lat] = point;
+        maxLng = Math.max(maxLng, lng);
+        minLng = Math.min(minLng, lng);
+        maxLat = Math.max(maxLat, lat);
+        minLat = Math.min(minLat, lat);
+      }
+    }
+
+    return { minLng, minLat, maxLng, maxLat };
+  }
+
+  private async getBoundingBoxFromGeometries<T extends Model>(
+    model: ModelStatic<T>,
+    whereCondition: WhereOptions,
+    geometryColumn: string,
+    entityType: EntityType,
+    identifier?: string
+  ): Promise<BoundingBoxDto> {
+    const envelopes = await model.findAll({
+      where: whereCondition,
+      attributes: [
+        [Sequelize.fn("ST_ASGEOJSON", Sequelize.fn("ST_Envelope", Sequelize.col(geometryColumn))), "envelope"]
+      ]
+    });
+
+    if (envelopes.length === 0) {
+      const errorMsg =
+        identifier !== undefined && identifier !== null && identifier !== ""
+          ? `No ${entityType.toLowerCase()} found with UUID ${identifier}`
+          : `No ${entityType.toLowerCase()} found with the provided criteria`;
+      throw new NotFoundException(errorMsg);
+    }
+
+    const { minLng, minLat, maxLng, maxLat } = this.extractBoundingBoxFromEnvelopes(envelopes);
+    return this.createBoundingBoxDto(minLng, minLat, maxLng, maxLat);
+  }
+
+  async getPolygonBoundingBox(polygonUuid: string): Promise<BoundingBoxDto> {
+    return this.getBoundingBoxFromGeometries(
+      PolygonGeometry,
+      { uuid: polygonUuid },
+      "geom",
+      EntityType.POLYGON,
+      polygonUuid
+    );
+  }
+
+  async getSiteBoundingBox(siteUuid: string): Promise<BoundingBoxDto> {
+    const site = await Site.findOne({
+      where: { uuid: siteUuid },
+      attributes: ["uuid"]
+    });
+
+    if (site === null || site === undefined) {
```

Comment:
Same thing here as in the cases in the controller; this should simply be `if (site == null) {` (and please check this service for any additional cases like this)